src/Components/MapComponent/MapComponent.jsx

```diff
@@ -227,64 +226,61 @@ class MapComponent extends PureComponent {
         dispatchSetCurrentStopsGeoJSON,
       } = this.props;
       let newHopIdx = -1;
-
-      // A segment is a linestring between to hops (also called via points).
-      // It's used to determine where to add the new hop.
-      let segments = features;
-
-      let currHop = null;
-      let multiLineString = null;
+      const { features } = this.initialRouteDrag;
       const { currentMot } = this.props;
 
       // In the case of the foot routing we can receive multiple line string between 2 hops (ex: one line string pro floor).
-      // So we have to re create the segment between 2 hops to be able to find the segment where to add the new hop.
+      // So we have to recreate the segment between 2 hops to be able to find the segment where to add the new hop.
       if (currentMot === 'foot') {
-        segments = [];
-        for (let i = 0; i < features.length; i += 1) {
-          const feature = features[i];
-          let hop = null;
-          if (feature.get('src')) {
-            hop = `${feature.get('src').join()}-${feature.get('trg').join()}`;
-          }
-          const clone = feature.getGeometry().clone();
-          if (currHop === hop || !hop) {
-            multiLineString.appendLineString(clone);
-          } else {
-            currHop = hop;
-            multiLineString = new MultiLineString(clone);
-            segments.push(new Feature(multiLineString));
-          }
+        // foot routing segments contain the start and end hop coordinates, we use them to find the hop index
+        const closestSegmentStartCoord = features[0].get('src');
+        const hop = this.markerVectorSource
+          .getFeatures()
+          .sort((a, b) => {
+            return a.get('idx') - b.get('idx');
+          })
+          .findIndex((f) => {
+            return (
+              to4326(f.getGeometry().getCoordinates()).join() ===
```

Comment:
can we use toLonLat  function?  form ol

Reply:
sure, we can... but `transform(coord, 'EPSG:3857', 'EPSG:4326')` should be the same as `toLonLat` right?
I think we can leave it like this for now

Reply:
yes  it just reimplements the wheel